test/e2e/addon/nvidiadeviceplugin.go

```diff
@@ -0,0 +1,59 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgo "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/aws/eks-hybrid/test/e2e/commands"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+)
+
+type NvidiaDevicePluginTest struct {
+	Cluster       string
+	K8S           clientgo.Interface
+	EKSClient     *eks.Client
+	K8SConfig     *rest.Config
+	Logger        logr.Logger
+	CommandRunner commands.RemoteCommandRunner
+
+	NodeName string
+}
+
+const (
+	nodeWaitTimeout          = 5 * time.Minute
+	nvidiaDriverWaitTimeout  = 20 * time.Minute
+	nvidiaDriverWaitInterval = 1 * time.Minute
+)
+
+// WaitForNvidiaDrivers checks if nvidia-smi command succeeds on the node
+func (n *NvidiaDevicePluginTest) WaitForNvidiaDriverReady(ctx context.Context) error {
```

Comment:
Does the plugin modify the node resource in k8s? I'm wondering if we would also need to check for gpu capacity in the node status.